PowervRA/Functions/Public/config-service/Invoke-vRAVAMIRestMethod.psm1

```diff
@@ -0,0 +1,139 @@
+function Invoke-vRAVAMIRestMethod {
+<#
+    .SYNOPSIS
+    Wrapper for Invoke-RestMethod with vRA VAMI specifics
+    
+    .DESCRIPTION
+    Wrapper for Invoke-RestMethod with vRA VAMI specifics
+
+    .PARAMETER Method
+    REST Method: GET or PUT
+
+    .PARAMETER URI
+    API URI, e.g. /nodes/list
+
+    .PARAMETER Body
+    REST Body in JSON format
+
+    .PARAMETER Headers
+    Optionally supply custom headers
+
+    .PARAMETER OutFile
+    Save the results to a file
+
+    .INPUTS
+    System.String
+    Switch
+
+    .OUTPUTS
+    System.Management.Automation.PSObject
+
+    .EXAMPLE
+    Invoke-vRARestMethod -Method GET -URI '/execute/command/run-prereq/node{nodeId}'
+
+    .EXAMPLE
+    $JSON = @"
+        {
+          "ValidationMode": "true",
+          "Components": "Website",
+          "ApplyFixes": "true"
+        }
+    "@
+
+    Invoke-vRARestMethod -Method PUT -URI '/execute/command/run-prereq/node{nodeId}' -Body $JSON
+#>
+[CmdletBinding()][OutputType('System.Management.Automation.PSObject')]
+
+    Param (
+
+    [parameter(Mandatory=$true)]
+    [ValidateSet("GET","PUT")]
+    [String]$Method,
+
+    [parameter(Mandatory=$true)]
+    [ValidateNotNullOrEmpty()]
+    [String]$URI,
+
+    [parameter(Mandatory=$false)]
+    [ValidateNotNullOrEmpty()]
+    [String]$Body,  
+    
+    [parameter(Mandatory=$false)]
+    [ValidateNotNullOrEmpty()]
+    [System.Collections.IDictionary]$Headers,
+
+    [parameter(Mandatory=$false)]
+    [ValidateNotNullOrEmpty()]
+    [String]$OutFile
+    
+    )   
+
+# --- Test for existing connection to vRA VAMI
+if (-not $Global:vRAVAMIConnection){
+
+    throw "vRA VAMI Connection variable does not exist. Please run Connect-vRAVAMI first to create it"
+}
+
+# --- Work with Untrusted Certificates
```

Comment:
You don't need to allow un trusted certs again here as it was already done for the session in Connect-vRAServer see what we did in PowervRO[https://github.com/jakkulabs/PowervRO/blob/development/PowervRO/Functions/Invoke-vRORestMethod.psm1](here) for an example. 

In fact we should probably remove this from Invoke-vRARestMethod too, i'll raise an issue to get it sorted,